client/components/TableOfContentsHighlight.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <div ref="wrapperRef" :class="['overflow-y-auto', props.wrapperClass]">
+    <slot />
+    <TableOfContentsHighlightSection
+      v-if="props.toc.sections"
+      :sections="props.toc.sections"
+      :depth="0"
+      :list-type-element="listTypeElement"
+      :active-id="activeId"
+      :handle-click="handleClick"
+      :make-toc-id="makeTocId"
+      :is-ssr="isSSR"
+      :list-class="props.listClass"
+      :nested-list-class="props.nestedListClass"
+      :list-item-class="props.listItemClass"
+      :list-item-active-class="props.listItemActiveClass"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+/**
+ * Table of Contents that highlights titles that are in the browser viewport
+ */
+import {
+  useTocActiveId,
+  useScrollTocContainer,
+  useValidateIds
+} from '../utilities/scroll'
+import type { RfcEditorToc } from '../utilities/tableOfContents'
+
+type Section = RfcEditorToc['sections'][number]
+
+type Props = {
+  toc: RfcEditorToc
+  listType: 'numbered' | 'ordered'
+  wrapperClass?: string
+  listClass?: string
+  nestedListClass?: string
+  listItemClass?: string
+  listItemActiveClass?: string
+}
+
+const props = defineProps<Props>()
+
+const listTypeElement = computed(() =>
+  props.listType === 'numbered' ? 'ol' : 'ul'
+)
+
+const wrapperRef = ref<HTMLElement>()
+
+const flattenSections = (section: Section): string[] =>
+  [section.id].concat(
+    section.sections ? section.sections.flatMap(flattenSections) : []
+  )
+
+const ids = computed(() => props.toc.sections.flatMap(flattenSections))
+
+if (
+  // safety check in case there's a coding error
+  // because if we pass non-strings to useActiveScroll then it will crash the page
+  ids.value.some((id) => typeof id !== 'string')
+) {
+  throw createError({
+    statusCode: 500,
+    statusMessage: `Server error. Please try again later. Toc ids were not string. Was: ${JSON.stringify(ids.value.map((id) => typeof id))}`,
```

Comment:
Do we have a way to log details and present a more generic message to the user? It's not a big deal here, but having a pattern like that might help avoid leaking internals.

Reply:
I don't think we have an approach for logging yet (eg NuxtKit logging, Sentry). It would be nice to be able to log errors clientside too.

And I'll make this only throw only on dev, not prod.